pkg/domain/scm_provider.go

```diff
@@ -0,0 +1,5 @@
+package domain
+
+type SCMProvider interface {
```

Comment:
Should we call it CIService instead of SCMProvider? I think it's better reflecting what it is. BitBucket, GitLab, GitHub are more CI tools than SCM in our case? And implementations would be BitBucketProvider, GitLabProvider, GitHubProvider in this case. Not sure myself what's the best naming, open for discussion.

Reply:
I prefer CI as a name too. Github and Bitbucket are SCM tools but *Github Actions* *Bitbucket Pipelines* *Jenkins* are CI tools and digger runs in CI systems hence CI is a better name in this case

Reply:
In this implementation, there is a package called `ci_runner` just for what you just mentioned. There is also the package `SCMProvider`, that handles the pull request work, publish a comment for example.

CI runner on the other side handles the event received (from the SCMProvider)

So you can have your SCM as Github, but your CI runner as Jenkins. Only Jenkins knows what kind of environment variables are there, specific to Jenkins. So we'll be handling events on Jenkins, but publishing comments on Github, hence the differentiation.

I don't know if that makes sense, or should we change it?